docs/static/persistent-queues.asciidoc

```diff
@@ -1,44 +1,228 @@
+Logstash attempts to deliver messages stored in the persistent queue until
+delivery succeeds at least once.  
+* Can absorb bursts of events without needing an external buffering mechanism like Redis or Apache Kafka.
+* Can be useful for Logstash deployments that need large buffers.
+You can enable persistent queues to buffer events on disk in a buffered publish-subscriber model without a message broker such as Redis, RabbitMQ, or
+Apache Kafka). 
+
+NOTE: Persistent queues are disabled by default.
+To enable them, check out <<configuring-persistent-queues>>.
 
 [[persistent-queues-limitations]]
-==== Limitations of Persistent Queues
+==== Limitations of persistent queues
+
+Persistent queues do not solve these problems:
+
+* Input plugins that do not use a request-response protocol cannot be protected from data loss. Tcp, udp, zeromq push+pull, and many other inputs do not have a mechanism to acknowledge receipt to the sender. 
+(Plugins such as beats and http, which *do* have an acknowledgement capability, are well protected by this queue.)
+* Data may be lost if an abnormal shutdown occurs before the checkpoint file has been committed.
+* A persistent queue does not handle permanent machine failures such as disk corruption, disk failure, and machine loss. 
+The data persisted to disk is not replicated.
+
+
+[[configuring-persistent-queues]]
+==== Configuring persistent queues
+
+To configure persistent queues, specify options in the Logstash <<logstash-settings-file,settings file>>:
+
+`queue.type`:: Specify `persisted` to enable persistent queues. By default, persistent queues are disabled (default: `queue.type: memory`).
+`path.queue`:: The directory path where the data files will be stored. By default, the files are stored in `path.data/queue`.
+`queue.type`:: Specify `persisted` to enable persistent queues. By default, persistent queues are disabled (default: `queue.type: memory`).
+`path.queue`:: The directory path where the data files will be stored. By default, the files are stored in `path.data/queue`. 
+`queue.page_capacity`:: The maximum size of a queue page in bytes. The queue data consists of append-only files called "pages". The default size is 64mb. Changing this value is unlikely to have performance benefits.
+`queue.drain`:: Specify `true` if you want Logstash to wait until the persistent queue is drained before shutting down. The amount of time it takes to drain the queue depends on the number of events that have accumulated in the queue. Therefore, you should avoid using this setting unless the queue, even when full, is relatively small and can be drained quickly. 
+// Technically, I know, this isn't "maximum number of events" it's really maximum number of events not yet read by the pipeline worker. We only use this for testing and users generally shouldn't be setting this.
+`queue.max_events`:: The maximum number of events that are allowed in the queue. The default is 0 (unlimited).
```

Comment:
Should we fold in the comments above into this setting? That users shouldn't really need to use this setting? Again, doesn't need to be done in this PR